### PR TITLE
Limit Health Checker goroutines

### PR DESCRIPTION
**Pull request by spectro30**

### CODE REVIEW

**Comment by spectro30** [comment]
**Merge this first**
https://github.com/kubedb/mariadb/pull/68

**Comment by spectro30** [comment]
e2e general test passed

File: pkg/controller/health.go [hunk 1]
```diff
@@ -53,206 +53,218 @@ func (c *Controller) RunHealthChecker(stopCh <-chan struct{}) {
...
+		klog.Errorf("Failed to list MariaDB objects with: %s", err.Error())
+		return
+	}
+
+	var wg sync.WaitGroup
+	for idx := range dbList {
+		db := dbList[idx]
+
+		// If the DB object is deleted, no need to perform health check.
+		if db.DeletionTimestamp != nil {
```
Comment (kamolhasan) [inline pkg/controller/health.go:81]: Also skip the health check for the halted dbs, `db.spec.halt==true`

File: pkg/controller/health.go [hunk 2]
```diff
@@ -53,206 +53,218 @@ func (c *Controller) RunHealthChecker(stopCh <-chan struct{}) {
...
+			continue
 		}
 
-		var wg sync.WaitGroup
-		for idx := range dbList {
-			db := dbList[idx]
+		wg.Add(1)
+		go func() {
```
Comment (kamolhasan) [inline pkg/controller/health.go:86]: Send DB object as input parameter,

go func(db *api.MariaDB){
... ...
... ...
}(db)

File: pkg/controller/health.go [hunk 3]
```diff
@@ -53,206 +53,218 @@ func (c *Controller) RunHealthChecker(stopCh <-chan struct{}) {
...
+							err = engine.Close()
+							if err != nil {
+								klog.Errorf("can't close the engine. error: %v,", err)
+							}
+						}
+					}()
+					isHostOnline, err := c.isHostOnline(db, engine)
 					if err != nil {
-						klog.Errorf("failed to update status for MariaDB: %s/%s", db.Namespace, db.Name)
+						klog.Warning("host is not online ", err.Error())
```
Comment (kamolhasan) [inline pkg/controller/health.go:127]: return from here, if an error occurs

**Review by kamolhasan — CHANGES_REQUESTED** [review]

File: pkg/controller/health.go (hunk 1 above)
Comment (spectro30) [reply pkg/controller/health.go:81]: added

File: pkg/controller/health.go (hunk 2 above)
Comment (spectro30) [reply pkg/controller/health.go:86]: added

File: pkg/controller/health.go (hunk 3 above)
Comment (spectro30) [reply pkg/controller/health.go:127]: fixed

**Pull request closed without merging**